### PR TITLE
[6.4.0] Do not allow applicable_licenses on platform.

### PR DESCRIPTION
**Pull request by comius**
RELNOTES:
`applicable_licenses` is no longer allowed on the `platform` rule. Additionally, `default_package_metadata` from any `package` rule will not be applied. 

PiperOrigin-RevId: 529808047
Change-Id: I7c2c682055bde6924d68858d38ef63f32457cdb9

### CODE REVIEW

**Comment by comius** [comment]
Hey @aiuto, I happens that I need this commit in 6.4.0 release.

To enable C++ toolchains, I need to add --platform flag to some of the builds.
And without this commit, Bazel <=6.3.x fails just after doing that.

(See https://github.com/bazelbuild/rules_android_ndk/pull/45/files)

**Review by aiuto — APPROVED** [review]

**Comment by aiuto** [comment]
Sorry for the slowness. I thought the 6.x. release manager approved all the cherrypicks.

**Comment by iancha1992** [comment]
> Sorry for the slowness. I thought the 6.x. release manager approved all the cherrypicks.

Hi @aiuto. Looks like @comius also requested a review from you. So if that's the case, then we would have to get your approval. Thanks for approving!

**Pull request merged**
